lib/authlogic/session/callbacks.rb

```diff
@@ -98,7 +98,20 @@ def #{method}
 
         def save_record(alternate_record = nil)
           r = alternate_record || record
-          r.save_without_session_maintenance(:validate => false) if r && r.changed? && !r.readonly?
+          return unless r
+
+          ensure_valid_new_record(r) if r.new_record?
+
+          if r.errors.blank? && r.changed? && !r.readonly?
+            r.save_without_session_maintenance(:validate => false)
+          else
+            r
+          end
```

Comment:
Awesome finding @tiegz It would be awesome if we can completely remove this "handling" because it feels weird for me also.